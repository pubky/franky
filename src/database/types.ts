// Primary keys for database entities
export type UserPK = string; // User's public key
export type PostPK = `${UserPK}:${string}`; // Format: userPK:postIdentifier

// Timestamp type used across the database
export type Timestamp = number;

// Sync status for database entities
export type SyncStatus = 'local' | 'homeserver' | 'nexus';

export type HomeserverActions = 'PUT' | 'DEL';
