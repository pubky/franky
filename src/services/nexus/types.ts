import { type UserPK, type PostPK, type Timestamp } from '@/database/types';

// Common types

export type NexusPostKind = 'short' | 'long' | 'repost' | 'reply' | 'link';

export interface NexusBookmark {
  created_at: number;
  updated_at: number;
}

export interface NexusUserLink {
  title: string;
  url: string;
}

// User types
export interface NexusUserDetails {
  name: string;
  bio: string;
  id: UserPK;
  links: NexusUserLink[] | null;
  status: string | null;
  image: string | null;
  indexed_at: Timestamp;
}

export interface NexusUserCounts {
  tagged: number; // the number of tags assign by this user to other entities ('users' or 'posts')
  tags: number; // the number of tags received by this user from other users
  unique_tags: number; // the number of unique tags received by this user from other users (distinct)
  posts: number;
  replies: number;
  following: number;
  followers: number;
  friends: number;
  bookmarks: number;
}

export interface NexusUserRelationship {
  following: boolean;
  followed_by: boolean;
  muted: boolean;
}

export interface NexusUserTag {
  label: string;
  taggers: UserPK[];
  taggers_count: number;
  relationship: boolean;
}

export interface NexusUser {
  details: NexusUserDetails;
  counts: NexusUserCounts;
  tags: NexusUserTag[];
  relationship: NexusUserRelationship;
}

// Post types
export interface NexusPostDetails {
  content: string;
  id: PostPK;
  indexed_at: number;
  author: UserPK;
  kind: NexusPostKind;
  uri: string;
  attachments: string[] | null;
}

export interface NexusPostCounts {
  tags: number;
  unique_tags: number;
  replies: number;
  reposts: number;
}

export interface NexusPostTag {
  label: string;
  taggers: UserPK[];
  taggers_count: number;
  relationship: boolean;
}

export interface NexusPostRelationships {
  replied: string | null;
  reposted: string | null;
  mentioned: UserPK[];
}

export interface NexusPost {
  details: NexusPostDetails;
  counts: NexusPostCounts;
  tags: NexusPostTag[];
  relationships: NexusPostRelationships;
  bookmark: NexusBookmark | null;
}

export interface NexusList {
  stream: string[];
  influencers: UserPK[];
  recommended: UserPK[];
}

export interface NexusBootstrapResponse {
  users: NexusUser[];
  posts: NexusPost[];
  list: NexusList;
}
