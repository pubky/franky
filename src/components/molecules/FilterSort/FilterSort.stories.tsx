import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { useState } from 'react';

import { FilterSort, type SortTab } from './FilterSort';

const meta: Meta<typeof FilterSort> = {
  title: 'Molecules/Filter/FilterSort',
  component: FilterSort,
  parameters: {
    layout: 'centered',
    docs: {
      description: {
        component: 'Preset for switching sort order between recent content and popularity.',
      },
    },
  },
  args: {
    selectedTab: 'recent',
  },
  argTypes: {
    selectedTab: {
      control: 'inline-radio',
      options: ['recent', 'popularity'],
    },
    onTabChange: { action: 'tab changed' },
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Playground: Story = {
  render: (args) => <FilterSort {...args} />,
};

export const Interactive: Story = {
  render: () => {
    const [selected, setSelected] = useState<SortTab>('recent');

    return <FilterSort selectedTab={selected} onTabChange={setSelected} />;
  },
  parameters: {
    docs: {
      description: {
        story: 'Interactive example showing the component controlling its own selected sort state.',
      },
    },
  },
};

export const PopularitySelected: Story = {
  args: {
    selectedTab: 'popularity',
  },
};
