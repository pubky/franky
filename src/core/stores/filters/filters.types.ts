// Filter constants
export const LAYOUT = {
  COLUMNS: 'columns',
  WIDE: 'wide',
  VISUAL: 'visual',
} as const;

export const SORT = {
  RECENT: 'recent',
  POPULARITY: 'popularity',
} as const;

export const REACH = {
  ALL: 'all',
  FOLLOWING: 'following',
  FRIENDS: 'friends',
  ME: 'me',
} as const;

export const CONTENT = {
  ALL: 'all',
  POSTS: 'posts',
  ARTICLES: 'articles',
  IMAGES: 'images',
  VIDEOS: 'videos',
  LINKS: 'links',
  FILES: 'files',
} as const;

// Filter types
export type LayoutType = (typeof LAYOUT)[keyof typeof LAYOUT];
export type SortType = (typeof SORT)[keyof typeof SORT];
export type ReachType = (typeof REACH)[keyof typeof REACH];
export type ContentType = (typeof CONTENT)[keyof typeof CONTENT];

export interface FiltersState {
  layout: LayoutType;
  sort: SortType;
  reach: ReachType;
  content: ContentType;
}

export interface FiltersActions {
  setLayout: (layout: LayoutType) => void;
  setSort: (sort: SortType) => void;
  setReach: (reach: ReachType) => void;
  setContent: (content: ContentType) => void;
  reset: () => void;
}

export type FiltersStore = FiltersState & FiltersActions;

// Initial state
export const filtersInitialState: FiltersState = {
  layout: LAYOUT.COLUMNS,
  sort: SORT.RECENT,
  reach: REACH.ALL,
  content: CONTENT.ALL,
};

// Action types for DevTools
export enum FiltersActionTypes {
  SET_LAYOUT = 'filters/setLayout',
  SET_SORT = 'filters/setSort',
  SET_REACH = 'filters/setReach',
  SET_CONTENT = 'filters/setContent',
  RESET = 'filters/reset',
}
