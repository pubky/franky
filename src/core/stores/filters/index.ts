export * from './filters.store';
export * from './filters.types';
export * from './filters.actions';
