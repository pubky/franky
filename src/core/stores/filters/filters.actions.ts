import { FiltersStore, FiltersActions, filtersInitialState, FiltersActionTypes } from './filters.types';
import { ZustandSet } from '../stores.types';

// Actions/Mutators - State modification functions
export const createFiltersActions = (set: ZustandSet<FiltersStore>): FiltersActions => ({
  setLayout: (layout) => {
    set({ layout }, false, FiltersActionTypes.SET_LAYOUT);
  },

  setSort: (sort) => {
    set({ sort }, false, FiltersActionTypes.SET_SORT);
  },

  setReach: (reach) => {
    set({ reach }, false, FiltersActionTypes.SET_REACH);
  },

  setContent: (content) => {
    set({ content }, false, FiltersActionTypes.SET_CONTENT);
  },

  reset: () => {
    set(filtersInitialState, false, FiltersActionTypes.RESET);
  },
});
