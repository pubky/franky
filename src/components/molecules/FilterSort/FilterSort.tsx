'use client';

import * as Atoms from '@/atoms';
import * as Libs from '@/libs';
import { SORT, type SortType } from '@/core/stores/filters/filters.types';

export type SortTab = SortType;

interface SortProps {
  selectedTab?: SortTab;
  onTabChange?: (tab: SortTab) => void;
}

const tabs: { key: SortTab; label: string; icon: React.ComponentType<{ className?: string }> }[] = [
  { key: SORT.RECENT, label: 'Recent', icon: Libs.SquareAsterisk },
  { key: SORT.POPULARITY, label: 'Popularity', icon: Libs.Flame },
];

export function FilterSort({ selectedTab = SORT.RECENT, onTabChange }: SortProps) {
  const handleTabClick = (tab: SortTab) => {
    onTabChange?.(tab);
  };

  return (
    <Atoms.FilterRoot>
      <Atoms.FilterHeader title="Sort" />

      <Atoms.FilterList>
        {tabs.map(({ key, label, icon: Icon }) => {
          const isSelected = selectedTab === key;

          return (
            <Atoms.FilterItem key={key} isSelected={isSelected} onClick={() => handleTabClick(key)}>
              <Atoms.FilterItemIcon icon={Icon} />
              <Atoms.FilterItemLabel>{label}</Atoms.FilterItemLabel>
            </Atoms.FilterItem>
          );
        })}
      </Atoms.FilterList>
    </Atoms.FilterRoot>
  );
}
