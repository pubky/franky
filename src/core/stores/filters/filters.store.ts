import { create } from 'zustand';
import { persist, devtools } from 'zustand/middleware';
import { FiltersStore, filtersInitialState } from './filters.types';
import { createFiltersActions } from './filters.actions';
import { withStoreBenchmarking } from '@/core/stores/storeBenchmarking';

// Store creation
export const useFiltersStore = create<FiltersStore>()(
  withStoreBenchmarking<FiltersStore>('filters-store')(
    devtools(
      persist(
        (set) => ({
          ...filtersInitialState,
          ...createFiltersActions(set),
        }),
        {
          name: 'filters-store',
          // Persist all filter states
          partialize: (state) => ({
            layout: state.layout,
            sort: state.sort,
            reach: state.reach,
            content: state.content,
          }),
        },
      ),
      {
        name: 'filters-store',
        enabled: process.env.NODE_ENV === 'development',
      },
    ),
  ),
);
