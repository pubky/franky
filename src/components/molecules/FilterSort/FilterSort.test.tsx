import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { FilterSort, type SortTab } from './FilterSort';
import { SORT } from '@/core/stores/filters/filters.types';

// Mock libs - use actual utility functions and icons from lucide-react
vi.mock('@/libs', async (importOriginal) => {
  const actual = await importOriginal<typeof import('@/libs')>();
  return { ...actual };
});

describe('FilterSort', () => {
  it('renders with default selected tab', () => {
    render(<FilterSort />);

    expect(screen.getByText('Sort')).toBeInTheDocument();
  });

  it('calls onTabChange when tab is clicked', () => {
    const mockOnTabChange = vi.fn();
    render(<FilterSort onTabChange={mockOnTabChange} />);

    const popularityElement = screen.getByText('Popularity');
    fireEvent.click(popularityElement);

    expect(mockOnTabChange).toHaveBeenCalledWith('popularity');
  });

  it('handles all tab types correctly', () => {
    const mockOnTabChange = vi.fn();
    render(<FilterSort onTabChange={mockOnTabChange} />);

    const tabs: SortTab[] = ['recent', 'popularity'];

    tabs.forEach((tab) => {
      const element = screen.getByText(tab === 'recent' ? 'Recent' : 'Popularity');

      fireEvent.click(element);
      expect(mockOnTabChange).toHaveBeenCalledWith(tab);
    });
  });

  it('handles tab switching correctly', () => {
    const mockOnTabChange = vi.fn();
    render(<FilterSort selectedTab={SORT.RECENT} onTabChange={mockOnTabChange} />);

    // Click on popularity tab
    const popularityElement = screen.getByText('Popularity');
    fireEvent.click(popularityElement);

    expect(mockOnTabChange).toHaveBeenCalledWith('popularity');
    expect(mockOnTabChange).toHaveBeenCalledTimes(1);
  });

  it('renders with different selected tabs', () => {
    const { rerender } = render(<FilterSort selectedTab="recent" />);

    let recentItem = screen.getByText('Recent').closest('[data-testid="filter-item"]');
    let popularityItem = screen.getByText('Popularity').closest('[data-testid="filter-item"]');

    expect(recentItem).toBeInTheDocument();
    expect(popularityItem).toBeInTheDocument();

    // Rerender with different selected tab
    rerender(<FilterSort selectedTab="popularity" />);

    recentItem = screen.getByText('Recent').closest('[data-testid="filter-item"]');
    popularityItem = screen.getByText('Popularity').closest('[data-testid="filter-item"]');

    expect(recentItem).toBeInTheDocument();
    expect(popularityItem).toBeInTheDocument();
  });
});

describe('FilterSort - Snapshots', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(<FilterSort />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Recent content selected tab', () => {
    const { container } = render(<FilterSort selectedTab={SORT.RECENT} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with Popularity content selected tab', () => {
    const { container } = render(<FilterSort selectedTab={SORT.POPULARITY} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
