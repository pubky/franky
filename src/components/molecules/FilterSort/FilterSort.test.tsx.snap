// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterSort > applies correct icon classes 1`] = `
<div
  class="mx-auto w-full flex flex-col gap-2"
  data-slot="filter-list"
  data-testid="filter-list"
>
  <p
    class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
    data-selected="true"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-square-asterisk w-5 h-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        width="18"
        x="3"
        y="3"
      />
      <path
        d="M12 8v8"
      />
      <path
        d="m8.5 14 7-4"
      />
      <path
        d="m8.5 10 7 4"
      />
    </svg>
    <span
      class=""
    >
      Recent
    </span>
  </p>
  <p
    class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-flame w-5 h-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
      />
    </svg>
    <span
      class=""
    >
      Popularity
    </span>
  </p>
</div>
`;

exports[`FilterSort > applies correct styling classes 1`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-square-asterisk w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="18"
      rx="2"
      width="18"
      x="3"
      y="3"
    />
    <path
      d="M12 8v8"
    />
    <path
      d="m8.5 14 7-4"
    />
    <path
      d="m8.5 10 7 4"
    />
  </svg>
  <span
    class=""
  >
    Recent
  </span>
</p>
`;

exports[`FilterSort > applies correct styling classes 2`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-flame w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
    />
  </svg>
  <span
    class=""
  >
    Popularity
  </span>
</p>
`;

exports[`FilterSort > renders with correct icons 1`] = `
<div
  class="mx-auto w-full flex flex-col gap-2"
  data-slot="filter-list"
  data-testid="filter-list"
>
  <p
    class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
    data-selected="true"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-square-asterisk w-5 h-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        width="18"
        x="3"
        y="3"
      />
      <path
        d="M12 8v8"
      />
      <path
        d="m8.5 14 7-4"
      />
      <path
        d="m8.5 10 7 4"
      />
    </svg>
    <span
      class=""
    >
      Recent
    </span>
  </p>
  <p
    class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
    data-selected="false"
    data-slot="filter-item"
    data-testid="filter-item"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-flame w-5 h-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
      />
    </svg>
    <span
      class=""
    >
      Popularity
    </span>
  </p>
</div>
`;

exports[`FilterSort > renders with custom selected tab 1`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-flame w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
    />
  </svg>
  <span
    class=""
  >
    Popularity
  </span>
</p>
`;

exports[`FilterSort > renders with default selected tab 1`] = `
<div
  class="mx-auto w-full flex flex-col gap-4 bg-background"
  data-slot="filter-root"
  data-testid="filter-root"
>
  <div
    class="flex flex-col gap-1"
    data-slot="filter-header"
    data-testid="filter-header"
  >
    <h2
      class="text-2xl text-muted-foreground font-light"
      data-testid="heading-2"
    >
      Sort
    </h2>
  </div>
  <div
    class="mx-auto w-full flex flex-col gap-2"
    data-slot="filter-list"
    data-testid="filter-list"
  >
    <p
      class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
      data-selected="true"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-square-asterisk w-5 h-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="3"
        />
        <path
          d="M12 8v8"
        />
        <path
          d="m8.5 14 7-4"
        />
        <path
          d="m8.5 10 7 4"
        />
      </svg>
      <span
        class=""
      >
        Recent
      </span>
    </p>
    <p
      class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
      data-selected="false"
      data-slot="filter-item"
      data-testid="filter-item"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-flame w-5 h-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
        />
      </svg>
      <span
        class=""
      >
        Popularity
      </span>
    </p>
  </div>
</div>
`;

exports[`FilterSort > renders with different selected tabs 1`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-square-asterisk w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="18"
      rx="2"
      width="18"
      x="3"
      y="3"
    />
    <path
      d="M12 8v8"
    />
    <path
      d="m8.5 14 7-4"
    />
    <path
      d="m8.5 10 7 4"
    />
  </svg>
  <span
    class=""
  >
    Recent
  </span>
</p>
`;

exports[`FilterSort > renders with different selected tabs 2`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-flame w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
    />
  </svg>
  <span
    class=""
  >
    Popularity
  </span>
</p>
`;

exports[`FilterSort > renders with different selected tabs 3`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-square-asterisk w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="18"
      rx="2"
      width="18"
      x="3"
      y="3"
    />
    <path
      d="M12 8v8"
    />
    <path
      d="m8.5 14 7-4"
    />
    <path
      d="m8.5 10 7 4"
    />
  </svg>
  <span
    class=""
  >
    Recent
  </span>
</p>
`;

exports[`FilterSort > renders with different selected tabs 4`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-flame w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
    />
  </svg>
  <span
    class=""
  >
    Popularity
  </span>
</p>
`;

exports[`FilterSort > shows correct visual state for selected and unselected tabs 1`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-foreground"
  data-selected="true"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-square-asterisk w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="18"
      rx="2"
      width="18"
      x="3"
      y="3"
    />
    <path
      d="M12 8v8"
    />
    <path
      d="m8.5 14 7-4"
    />
    <path
      d="m8.5 10 7 4"
    />
  </svg>
  <span
    class=""
  >
    Recent
  </span>
</p>
`;

exports[`FilterSort > shows correct visual state for selected and unselected tabs 2`] = `
<p
  class="cursor-pointer flex gap-2 items-center text-base font-medium transition-colors text-muted-foreground hover:text-foreground/80"
  data-selected="false"
  data-slot="filter-item"
  data-testid="filter-item"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-flame w-5 h-5"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
    />
  </svg>
  <span
    class=""
  >
    Popularity
  </span>
</p>
`;
